src/main/java/rs/teslaris/core/controller/MergeController.java

```diff
@@ -0,0 +1,105 @@
+    }
+
+    @PatchMapping("/employment/{sourceOUId}/target/{targetOUId}/person/{personId}")
+    @PreAuthorize("hasAuthority('MERGE_OU_EMPLOYMENTS')")
+    @ResponseStatus(HttpStatus.NO_CONTENT)
+    public void switchEmployeeToOtherOrganisationUnit(@PathVariable Integer sourceOUId,
+                                                      @PathVariable Integer targetOUId,
+                                                      @PathVariable Integer personId) {
+        mergeService.switchPersonToOtherOU(sourceOUId, targetOUId, personId);
+    }
+
+    @PatchMapping("/employment/{sourceOUId}/target/{targetOUId}")
+    @PreAuthorize("hasAuthority('MERGE_OU_EMPLOYMENTS')")
+    @ResponseStatus(HttpStatus.NO_CONTENT)
+    public void switchAllEmployeesToOtherOrganisationUnit(@PathVariable Integer sourceOUId,
+                                                          @PathVariable Integer targetOUId) {
+        mergeService.switchAllPersonsToOtherOU(sourceOUId, targetOUId);
+    }
+
+    @PatchMapping("/conference/{targetConferenceId}/proceedings/{proceedingsId}")
+    @PreAuthorize("hasAuthority('MERGE_CONFERENCE_PROCEEDINGS')")
+    @ResponseStatus(HttpStatus.NO_CONTENT)
+    public void switchProceedingsToOtherConference(@PathVariable Integer targetConferenceId,
+                                                   @PathVariable Integer proceedingsId) {
+        mergeService.switchProceedingsToOtherConference(targetConferenceId, proceedingsId);
+    }
+
+    @PatchMapping("/conference/{sourceConferenceId}/target/{targetConferenceId}")
+    @PreAuthorize("hasAuthority('MERGE_CONFERENCE_PROCEEDINGS')")
+    @ResponseStatus(HttpStatus.NO_CONTENT)
+    public void switchAllProceedingsToOtherConference(@PathVariable Integer sourceConferenceId,
+                                                      @PathVariable Integer targetConferenceId) {
+        mergeService.switchAllProceedingsToOtherConference(sourceConferenceId, targetConferenceId);
+    }
+
+    @PatchMapping("/proceedings/{targetProceedingsId}/publication/{publicationId}")
+    @PreAuthorize("hasAuthority('MERGE_PROCEEDINGS_PUBLICATIONS')")
+    @ResponseStatus(HttpStatus.NO_CONTENT)
+    public void switchProceedingsPublicationToOtherProceedings(
+        @PathVariable Integer targetProceedingsId,
+        @PathVariable Integer publicationId) {
+        mergeService.switchProceedingsPublicationToOtherProceedings(targetProceedingsId,
+            publicationId);
+    }
+
+    @PatchMapping("/proceedings/{sourceProceedingsId}/target/{targetProceedingsId}")
+    @PreAuthorize("hasAuthority('MERGE_PROCEEDINGS_PUBLICATIONS')")
+    @ResponseStatus(HttpStatus.NO_CONTENT)
+    public void switchAllProceedingsPublicationsToOtherProceedings(
+        @PathVariable Integer sourceProceedingsId,
+        @PathVariable Integer targetProceedingsId) {
+        mergeService.switchAllPublicationsToOtherProceedings(sourceProceedingsId,
+            targetProceedingsId);
+    }
```

Comment:
Why switchInvolvementsToOtherPerson is not a method in this class?

Reply:
I don't know, I wanted to discuss the placement of these methods on the meeting but I forgot :D Should I switch them to MergeController? Should Service methods also be switched to MergeService?

Reply:
I would suggest to put them all together into MergeController (and also to use MergeService for all service methods of this type)

Reply:
Refactored :smile: